flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/CheckpointCoordinator.java

```diff
@@ -183,6 +189,15 @@ public class CheckpointCoordinator {
     /** A handle to the current periodic trigger, to cancel it when necessary. */
     private ScheduledFuture<?> currentPeriodicTrigger;
 
+    /**
+     * The timestamp (via {@link Clock#relativeTimeMillis()}) when the next checkpoint will be
+     * triggered.
+     *
+     * <p>If it's value is {@link Long#MAX_VALUE}, it means there is not a next checkpoint
+     * scheduled.
+     */
+    private volatile long nextCheckpointTriggeringRelativeTime;
```

Comment:
Would it be useful to mark this variable with `@GuardedBy("lock")`?

Reply:
`nextCheckpointTriggeringRelativeTime` and `currentPeriodicTrigger` should be updated together, so I added `synchronized(lock){}` around code blocks related to these variables instead of adding `@GuardedBy("lock")` to each of them.

Reply:
As discussed offline, we agree it is useful to annotate this variable with `@GuardedBy("lock")`.